### DISCUSSION TITLE
Material properties and sharp transitions

### DISCUSSION THREAD

**Discussion by gka80** [idaholab/moose#16189] [Q&A General] [answered]
I currently have a material property defined using a DerivativeParsedMaterial with an 'if' statement. The 'if' statement acts as a switch for the material property (i.e. similar to a first-order phase transition). I've tested the model above and below the critical point and it behaves as expected. However, because of the way I've modeled this, there's a sharp transition at the critical point. I'd like to "help the solver out" around this transition point in the hopes of improving convergence around this point and reducing runtime.

What's a good way to accomplish this? If something doesn't exist, I don't have a problem with contributing if someone has an idea.

Thank you,

-Garrett

**Comment by WilkAndy** [comment, marked as answer]
I think this depends a bit on your physics.  For instance, in my fields where the finite element is frequently a coarse-scale representation of some complicated (unknown?) micro physics, a "mollifier" would be appropriate here.  That is, replace if(x<0, -1, 1) with tanh(x/m), where m is the mollifier (or some similar function).

a

________________________________

**Reply by gka80** [reply]
Aha! That seems to have done the trick to get it over the "hump". Thank you, Andy.

Just in case anyone else comes across this, I defined a "switching" function, f(x) = 0.5+0.5*tanh((x-x_c)/w), where 'x_c' represents the critical point (i.e. a temperature) and 'w' represents the "transition width." If 'p_below(x)' is the property below the critical point and 'p_above(x)' is the property above the critical point, you can transition between these two properties by: p(x) = p_below(x)*(1-f(x))+p_above(x)*f(x).
